src/misc/lv_anim.h

```diff
@@ -133,6 +139,8 @@ struct _lv_anim_t {
     lv_anim_custom_exec_cb_t custom_exec_cb;/**< Function to execute to animate,
                                                  same purpose as exec_cb but different parameters*/
     lv_anim_start_cb_t start_cb;         /**< Call it when the animation is starts (considering `delay`)*/
+    lv_anim_pause_cb_t pause_cb;    /**< Call it when the animation is paused */
+    lv_anim_resume_cb_t resume_cb;  /**< Call it when the animation is resumed */
     lv_anim_completed_cb_t completed_cb; /**< Call it when the animation is fully completed*/
     lv_anim_deleted_cb_t deleted_cb;     /**< Call it when the animation is deleted*/
```

Comment:
In most of the cases the users want the animation to be deleted automatically on competition. However I do understand for better life cycle management you might need the animation pointers. Can you mention a use case when the animation is not deleted automatically with the widgets they are animating, but you are managing the animation separately?

Reply:
> a use case when the animation is not deleted automatically

What about a animation when users clicks screen over and over? Reuse the previous animation already setup sounds reasonable.

The complete life cycle management actually comes from the binding language, where the memory should be free'd at certain time. For example, `lua` code may still hold a reference to `anim` handler while it's already automatically deleted.

Reply:
Reusing the animation is possible as of the PR, but see my comment making it more streamlined above.

---

I don't clearly see the relevance of the problems raised by the binding language.
- Freeing at certain time is safe and possible by `lv_anim_delete()` (Right now `lv_anim_delete_by_ptr()` can lead to double freeing if the animation has finished beforehand, `lv_anim_tigger_reusable()` would solve that.)
- Is "holding a reference to deleted handlers" truly animation specific? I'd imagine the same thing would happen with a widget that is being closed, or any other lvgl object, that is being freed by internal mechanisms, like patent-child "garbage collection".

I'm not so familiar with lua. Would you mind creating an example script where this issue is demonstrated? It would be a big help.